conda_smithy/schema.py

```diff
@@ -0,0 +1,1240 @@
+        Azure pipeline artifact",
+    )
+
+    timeout_minutes: Optional[Union[int, Nullable]] = Field(
+        default=None,
+        description="The maximum amount of time (in minutes) that a \
+            job can run before it is automatically canceled",
+    )
+
+
+class GithubConfig(BaseModel):
+    model_config: ConfigDict = ConfigDict(extra="forbid")
+
+    user_or_org: Optional[str] = Field(
+        description="The name of the GitHub user or organization, \
+        if passed with the AzureConfig provider, must comply with the value of the user_or_org field",
+        default="conda-forge",
+    )
+    repo_name: Optional[str] = Field(
+        description="The name of the repository",
+        default="",
+    )
+    branch_name: Optional[str] = Field(
+        description="The name of the branch to execute on",
+        default="main",
+    )
+    tooling_branch_name: Optional[str] = Field(
+        description="The name of the branch to use for rerender+webservices \
+            github actions and conda-forge-ci-setup-feedstock references",
+        default="main",
+    )
+
+
+class GithubActionsConfig(BaseModel):
+    model_config: ConfigDict = ConfigDict(extra="forbid")
+
+    artifact_retention_days: Optional[int] = Field(
+        description="The number of days to retain artifacts",
+        default=14,
+    )
+
+    cancel_in_progress: Optional[bool] = Field(
+        description="Whether to cancel jobs in the same build if one fails.",
+        default=True,
+    )
+
+    max_parallel: Optional[Union[int, Nullable]] = Field(
+        description="The maximum number of jobs to run in parallel",
+        default=None,
+    )
+
+    self_hosted: Optional[bool] = Field(
+        description="Whether to use self-hosted runners",
+        default=False,
+    )
+
+    store_build_artifacts: Optional[bool] = Field(
+        description="Whether to store build artifacts",
+        default=False,
+    )
+
+    timeout_minutes: Optional[int] = Field(
+        default=360,
+        description="The maximum amount of time (in minutes) that a \
+            job can run before it is automatically canceled",
+    )
+
+    triggers: Optional[list] = Field(
+        default=[],
+        description="Triggers for Github Actions. Defaults to push, pull_request, \
+            when not self-hosted and push when self-hosted",
+    )
+
+    upload_packages: Optional[bool] = Field(
+        default=True,
+        description="Whether to upload the packages to Anaconda.org. Useful for testing.",
+    )
+
+
+class BotConfigVersionUpdates(BaseModel):
+    """
+    This dictates the behavior of the conda-forge auto-tick bot for version
+    updates
+    """
+
+    model_config: ConfigDict = ConfigDict(extra="forbid")
+
+    random_fraction_to_keep: Optional[float] = Field(
+        None,
+        description="Fraction of versions to keep for frequently updated packages",
+    )
+
+    exclude: Optional[List[Union[str, float]]] = Field(
```

Comment:
That list is awesome, thanks!

Reply:
@isuruf, any thoughts here? Since it's only three feedstocks, I think it's ok to be strict here and only allow `str` (same with `abi_migration_branches`).

Reply:
The errors are a bit cryptic here if I remember correctly. That's why I added these. I don't mind changing it. Let's send PRs to those 3 feedstocks.

Reply:
Added:

- https://github.com/conda-forge/cuda-python-feedstock/pull/77
- https://github.com/conda-forge/wgpu-native-feedstock/pull/18
- https://github.com/conda-forge/aws-c-common-feedstock/pull/94